### DISCUSSION TITLE
whisper not using CUDA issue

### DISCUSSION THREAD

**Discussion by joer33304** [openai/whisper#415] [Q&A] [answered]
I installed whisper and pytorch via pip. It run super slow and torch.cuda.is_available() showed false. Could not get that to show true via any help using pip. I uninstalled it and re installed via conda. Now it shows true but Anaconda seems only to run in its own shell where it can't find whisper.

Below are my data. My CUDA is 1.18 but torch seems to have only latest version 1.16. However I doubt that this is the issue as it works when installing with conda. How do I get this to run ?

nvidia-smi CUda 11.8
nvcc --version Cuda 11.8
Pyhton version  3.10.8
pip3 install --pre torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/nightly/cu116
Successfully installed torch-1.12.1

C:\Windows\system32>python
Python 3.10.8 (tags/v3.10.8:aaaf517, Oct 11 2022, 16:50:30) [MSC v.1933 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> x = torch.rand(5, 3)
>>> print(x)
tensor([[0.8495, 0.4900, 0.3638],
        [0.5326, 0.1098, 0.0787],
        [0.0755, 0.5029, 0.1658],
        [0.4929, 0.7832, 0.8566],
        [0.5083, 0.6562, 0.4616]])
>>> torch.cuda.is_available()
False

**Comment by Jxspa** [comment, marked as answer]
This worked for me.
```
pip uninstall torch
pip install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu116
```

**Reply by tophtab** [reply]
THX, it works

**Reply by martinkirov** [reply]
thank you so much!!

**Reply by Immrspy** [reply]
This fix was still necessary for me to run whisper on Windows 10, thanks for the help!

**Reply by dmurphy779** [reply]
Works! - This should be in the documentation.

**Reply by Kaoru8** [reply]
Unfortunately, this didn't work for me. What eventually did is the following - open a conda terminal, activate your whisper environment, then run:
```
conda uninstall torch
python -m pip uninstall torch
python -m pip install torch==1.13.1+cu116 torchvision==0.14.1+cu116 torchaudio==0.13.1 --extra-index-url https://download.pytorch.org/whl/cu116
```

**Reply by AstroGD** [reply]
@Kaoru8 Thank you so much! Had the exact same issue and the solution of @Jxspa didn't work for me as well but yours did

**Comment by joer33304** [comment]
thanks it finally worked by using cu116, I was always trying to use a
higher version

**Reply by dhalmy** [reply]
had the same exact issue

**Comment by jlopp** [comment]
I hit this same issue when I tried setting up Whisper on a Windows machine after using it on Linux for a long time. Took me hours to figure out the fix because I didn't know about this discussion area. It sure would be nice if this could be fixed to install a CUDA compatible version of pytorch automatically!

**Reply by mmtmn** [reply]
For anyone using Windows, you can't use the Python from the Microsoft store to set up CUDA. Make sure to uninstall python if you download it off the Microsoft store and follow along the traditional installation guides..

**Comment by mmtmn** [comment]
Isn't there some paramater we can use that forces the system to use cuda when its already working with other AIs without a problem? something like: model.transcribe("audio.mp3", gpu)
